resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -207,4 +207,28 @@ package object config {
         """.stripMargin)
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("1s")
+
+  private[spark] val DRIVER_EXPOSE_INGRESS =
+    ConfigBuilder("spark.kubernetes.driver.exposeIngress")
+      .doc(
+        """
+          | When initially contacting the driver, use an Ingress when the submitting client
+          | passes application dependencies to the driver pod. The Ingress also remains and
+          | exposes the Spark UI. This requires an Ingress controller must be installed on
+          | the cluster.
+        """.stripMargin
+      )
+      .booleanConf
+      .createWithDefault(false)
+
+  private[spark] val INGRESS_BASE_PATH =
```

Comment:
I believe so. I was wondering if we could rely on the `LoadBalancerStatus` we get back from querying API server for the Ingress status.